Confirm when leaving team

muhammad-ammar opened the pull request:
TNL-3020

Ticket: https://openedx.atlassian.net/browse/TNL-3020
Sandbox: http://tnl3020.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course

muhammad-ammar commented:
@frrrances @marcotuts I have tried to move the styling into common and lms but things are not working expected. Also there are [conflicts](https://github.com/edx/edx-platform/pull/9265/files#diff-24805496f6acfff7dab33e1c8de33470R56)
Also [here](https://github.com/edx/edx-platform/pull/9265/files#diff-cca469339b66739c7789971ef8c55f0dR421) if i remove `!optional` sass compile will fail because it is unable to find `%ui-btn-secondary`

I have also tried to move only those styles that are being used by confirmation dialog but no success.

Can you see the changes and suggest the best way to move the styling needed for confirmation dialog into common so that lms and studio both can use it?

muhammad-ammar commented:
jenkins run bokchoy

muhammad-ammar commented on lms/static/sass/base/_variables.scss:
```diff
@@ -144,8 +210,8 @@ $linkedin-blue: #0077B5;
-$blue: rgb(0, 120, 176);
-$yellow: rgb(255, 252, 221);
+//$blue: rgb(0, 120, 176);
+//$yellow: rgb(255, 252, 221);
```
@marcotuts What to do with these?

marcotuts replied:
for blue we'll want to keep the LMS blue instead of Studio, since that has been reviewed for a11y contrast. In fact, given your change above we should be able to remove this duplicated blue line. 

As I noted above, I will double check the yellow question with the team to get thoughts on that.

muhammad-ammar commented on lms/static/sass/base/_variables.scss:
```diff
@@ -134,6 +134,72 @@ $green-u1: desaturate($green,15%);
 
+// copied from cms/static/sass/_variables.scss
+$yellow: rgb(255, 252, 221);      // yellow color used by LMS
+//$yellow: rgb(237, 189, 60);     // yellow color used by Studio
```
@marcotuts Which color should be use?

marcotuts replied:
Curiously, neither is the right yellow according to our recently updated pattern library work. :) (http://ux.edx.org/elements/colors/) I will talk to the UX team and get their thoughts today. thanks for flagging this.

muhammad-ammar commented on lms/static/sass/base/_variables.scss:
```diff
@@ -134,6 +134,72 @@ $green-u1: desaturate($green,15%);
+
+// copied from cms/static/sass/_variables.scss
+$blue: rgb(0, 120, 176);    // blue color used by LMS
+//$blue: rgb(0, 159, 230);  // blue color used by Studio
```
@marcotuts Which color should be use?

marcotuts replied:
We should use the LMS blue here instead of studio. It is a bit odd that it varies across applications currently, but we can clean that up eventually. 

For the colors section, I think it is also ok to drop the // copied from... note.

muhammad-ammar commented on lms/static/sass/elements/_controls.scss:
```diff
@@ -416,3 +416,83 @@
   }
+
+  %btn-secondary-white {
+  @extend %ui-btn-secondary !optional;
```
@marcotuts made `!optional` otherwise need to copy dependancies from studio, also `btn-secondary-white` is not being used in dialog.

muhammad-ammar commented on lms/static/sass/elements/_system-feedback.scss:
```diff
@@ -145,9 +145,955 @@
+  .action-dismiss {
+
+    .button {
+      @extend %btn-secondary-white !optional;
```
@marcotuts made `!optional` otherwise need to copy dependancies from studio, also `btn-secondary-white` is not being used in dialog.

marcotuts replied:
sounds great. would you mind adding a  comment to this effect here to clarify why optional was used? I can take a look in a later PR about moving the dependency over to LMS as well. thanks!

muhammad-ammar commented on lms/static/sass/elements/_system-feedback.scss:
```diff
@@ -145,9 +145,955 @@
+    background-color: $gray-d1;
+
+    .actions-list {
+      @extend %actions-list !optional;
```
@marcotuts made `!optional` otherwise need to copy dependancies from studio, also `btn-secondary-white` is not being used in dialog.

marcotuts replied:
Is the only dependency here the %actions-list extend from cms/.. _controls.scss? If so this seems like a reasonable dependency to move to the _controls.scss file for the lms.

muhammad-ammar replied:
@marcotuts `actions-list` has rules for `drag-handle` class which uses `cms/static/images/drag-handles.png`, so i am not sure what should i do here.

marcotuts replied:
Ah ok, the optional flag makes sense here then, thanks.

muhammad-ammar commented on lms/static/sass/_build-lms.scss:
```diff
@@ -76,6 +75,10 @@
 
+@import 'mixins';
+@import 'mixins-inherited';
+@import 'elements/system-feedback';
```
@marcotuts if we add these above in the start of file then we will get conflicts with a burbon mixin named as `button.`

muhammad-ammar commented:
@marcotuts currently cancel button and title styling is incorrect, i tried to fix but no luck.

muhammad-ammar commented:
jenkins run js

muhammad-ammar commented:
jenkins run js

muhammad-ammar commented:
@symbolist Can you review the changes. Thanks

muhammad-ammar commented:
@marcotuts Can you review the changes. Thanks

muhammad-ammar commented:
@catong Can you review the UI strings? Thanks

catong commented on lms/djangoapps/teams/static/teams/js/views/team_profile.js:
```diff
@@ -72,20 +73,27 @@
-                                });
+                    ViewUtils.confirmThenRunOperation(
+                        gettext('Leave this team?'),
+                        gettext('Leaving a team means you can no longer post on this team, and your spot is opened for another learner.'),
```
"If you leave, you can no longer post in this team's discussions. Your place will be available to another learner."

muhammad-ammar commented:
@catong Updated the text. Thanks

muhammad-ammar commented:
@marcotuts I am stuck. Dialog styling is not working as expected. Dialog rules are being overridden. Can you help? Thanks

symbolist commented:
Hi @cptvitamin, this PR adds a confirmation step when a user tries to leave a team. This required moving the dialog box that is used in Studio to common. It does not look like the dialog box has been reviewed for a11y. So we would like a review of that. The template file is `system-feedback.underscore`.

Also, your earlier comments about the sidebar of the team profile page that are noted on https://openedx.atlassian.net/browse/TNL-1930 have also been addressed. These changes have also been deployed to the sandbox. Can you check if everything is behaving as expected?

Thanks!

marcotuts commented:
@muhammad-ammar  Thanks for all the work on this. I was able to review the sandbox and have two suggested styling changes. in _shame.scss I would add a section for  // LMS system feedback button overrides. This section is necessary due to base LMS button styles. 

The goal for these overrides is to reset to styles for LMS messages. using .is-in-course .wrapper-prompt .action-secondary should I believe be the right class.  

Two rules to add are - 
box-shadow: none;
text-shadow: none; 

This will remove the odd button styling for the Cancel action in LMS version of the prompt. The heading uses the LMS all caps h2 styling, which is fine to keep as is for now. 

Those were the primary visual issues I noted, thanks again for the effort to move this out separate from studio! 

:+1: after styling overrides.

muhammad-ammar commented:
@marcotuts I have added style overrides. Also made some other little style changes. Please have a look. Thanks!

symbolist commented:
Hi @clrux, will you be able to look at this? I added some details in the comment to @cptvitamin above. But he will not be able to look at this.

catong commented:
@muhammad-ammar LGTM for UI text strings. Thanks!

clrux commented:
@muhammad-ammar A few thoughts here:

* the dialog confirmation box works nicely, but I think @talbs or @frrrances should chime in on its use, as it's primarily a Studio element.
* while the dialog works with screen readers, there's a funky set of options that aren't "actionable", meaning I can't do anything with them even though they're in the markup
![screen shot 2015-08-24 at 2 00 41 pm](https://cloud.githubusercontent.com/assets/2112024/9448124/e8b58600-4a68-11e5-8eb1-a51d406e7274.png)

muhammad-ammar commented:
jenkins run lettuce

muhammad-ammar commented:
jenkins run all

muhammad-ammar commented:
jenkins run all

muhammad-ammar commented:
jenkins run all

muhammad-ammar commented:
jenkins run all

muhammad-ammar commented:
@marcotuts I have made style override changes along with some changes due to replacing `<a>` with `<button>`. Please have a final look. Thanks!
@clrux Addressed your feedback. Please continue the review. Thanks!

muhammad-ammar commented:
> Hi @cptvitamin, this PR adds a confirmation step when a user tries to leave a team. This required moving the dialog box that is used in Studio to common. It does not look like the dialog box has been reviewed for a11y. So we would like a review of that. The template file is system-feedback.underscore.
Also, your earlier comments about the sidebar of the team profile page that are noted on https://openedx.atlassian.net/browse/TNL-1930 have also been addressed. These changes have also been deployed to the sandbox. Can you check if everything is behaving as expected?
Thanks!

@clrux  ^^

clrux commented:
The buttons are more clear, thanks. I'd like for @frrrances or @talbs to weigh in on the use of this Studio modal in LMS though.

clrux commented:
@muhammad-ammar Regarding TNL-1930, when the "New Post" button is clicked, please send focus directly to the expanded post form. Additionally, when "Expand Discussion" is clicked, focus should be sent to the top of the expanded posts.

cahrens commented:
@andy-armstrong Can you take a look at this PR?

andy-armstrong commented on cms/static/js/views/feedback.js:
```diff
@@ -1,146 +0,0 @@
-define(["jquery", "underscore", "underscore.string", "backbone", "js/utils/templates"],
```
Can you not delete these files but instead move them from cms to common. I don't want to lose the history.

andy-armstrong replied:
It also makes it harder to see what changes (if any) have been made.

muhammad-ammar replied:
@andy-armstrong I don't know what happened here. I have used `git mv`. I am looking into this. Thanks!

cahrens commented:
@clrux Regarding your comment about sending focus to the expanded post form, isn't this the same issue as https://openedx.atlassian.net/browse/TNL-2620? Accessibility fixes to the inline discussion component are beyond the scope of the Teams work-- I think that both the "New Post' and "Expand Discussion" issues are generic inline discussion component issues and are not related to the new Teams UI.

andy-armstrong commented on common/static/common/js/spec/components/.gitignore:
```diff
@@ -0,0 +1 @@
+*.js
```
What's the purpose of this? Why do we want Git to ignore these files? I'm not sure that this should be checked in.

muhammad-ammar replied:
@andy-armstrong I have moved `feedback_spec.coffee` here, so during tests `coffee` is compiled into `.js` and we don't want to commit this js file so i am gitignoring it.

andy-armstrong replied:
@muhammad-ammar I don't like this idea, because it will make it harder for people to notice that they haven't added new JS files to Git. I recommend either converting this file to JavaScript, or creating a top level coffee directory. My preference would be the former if possible.

andy-armstrong commented on lms/djangoapps/teams/static/teams/js/spec/views/team_profile_spec.js:
```diff
@@ -111,7 +121,7 @@ define([
 
                 expect(view.$('.new-post-btn').length).toEqual(1);
-                clickLeaveTeam(requests, view);
+                clickLeaveTeam(requests, view, true);
```
Nit: I'd prefer this be passed as an options object so that it is easier to see what the ``true`` means.

    clickLeaveTeam(requests, view, {cancel: true});

symbolist commented:
@explorerleslie If you would like to look at this.

clrux commented:
@cahrens Ah, fair point. Then disregard!

andy-armstrong commented on lms/static/sass/_build-lms.scss:
```diff
@@ -76,6 +75,10 @@
 // overrides
 @import 'developer'; // used for any developer-created scss that needs further polish/refactoring
-@import 'shame';     // used for any bad-form/orphaned scss
+@import 'shame';     // used for any bad-form/orphaned scss
```
Nit: missing newline

symbolist commented:
@clrux Just want to confirm – otherwise have the issues noted in https://openedx.atlassian.net/browse/TNL-1930 been addressed?

andy-armstrong commented:
@muhammad-ammar This looks great! I can't wait to use it in other parts of the LMS, e.g. in the forums rewrite. :+1: from me, but if possible please try to preserve the history for the files that are moving into common.

explorerleslie commented:
@symbolist the page doesn't refresh after you leave the team for me. Have you seen this? I even created a new user in case it was that weird issue with multiple people using the same account at the same time. This is confusing because it does actually remove you from the team, but it doesn't refresh. So then when you click Leave Team again, the page errors.

muhammad-ammar commented:
@explorerleslie I am not seeing this issue. Could you please share the steps to reproduce the issue. Thanks!

muhammad-ammar commented:
@andy-armstrong Addressed your feedback. Please continue the review. 

Regarding history preservation of moved files, locally i can see the complete history of moved files. I am not sure about the github behavior about showing moved files. Some of the moved files are shown correctly as renamed on github but some are not.

symbolist commented:
@explorerleslie Could the issue you are seeing have something to do with: https://openedx.atlassian.net/browse/TNL-3038?

explorerleslie commented:
@symbolist YES. I hate hate hate hate hate SOPHOS. Moving on...

andy-armstrong commented:
:+1:

clrux commented:
:+1:

muhammad-ammar commented:
jenkins run bokchoy

muhammad-ammar commented:
jenkins run bokchoy

symbolist commented:
Looks good to go! :+1:

jzoldak commented:
@muhammad-ammar @symbolist @cahrens 
The JavaScript.firefox.Library Instructor Access Page read-write access tests have gone flaky, and it seems to be correlated to this PR being merged in.
Submitted as ticket [TNL-3166](https://openedx.atlassian.net/browse/TNL-3166)
